### PR TITLE
Fixes powernet random event magically refilling SMES and setting their output to M A X I M U M P O W E R

### LINKED ISSUE #11704: Electrical storm event refills SMES's
(WEB REPORT BY: dragonbro REMOTE: 172.93.109.202:7777)

> Revision (Should be above if you're viewing this from ingame!)
> 44491382b2416afb61430af5d0f6834d8206637d 
> 
> General description of the issue
> For some strange reason, when the electrical storm fires and does its thing, previously near-empty SMES's come out of it completely refilled, despite not having any power sources
> 
> What you expected to happen
> The turned off power to not produce more power
> 
> What actually happened
> FREE POWER
> 
> Steps to reproduce if possible
> Have a shit engineering team
> Watch as they die in stupid ways, not getting power set up
> An electrical storm rolls in and out, turning station power on and off as it goes
> Marvel that SMES's are now at full, enjoy another 40 minutes to get the station's power problems sorted

### PR DESCRIPTION
**Pull request by 9600bauds**
Before:
* APCs magically go poof losing all their charge and setting all settings to "off", you can jam a new cell in tho
* SMES magically go poof losing all their charge and set all settings to "off", you can still just turn them back on though but it won't really help since
* when power is restored all APCs magically get FULL charge back and have all settings turned to ON even if they were set to off before the event
* when power is restored all SMES magically get !!FULL FUCKING CHARGE!! and are set to !!200K FUCKING OUTPUT!!

Now:
* APCs still magically go poof and lose their charge, you can still take the cell out and jam a new cell in (the old one will never regain power if you do so), when regaining power they go back to their old settings and go to their old charge % (or stay at their current if you swapped the cell, whichever is higher). I left it like this because I prefer this functioning over not being able to swap out the cell, which would make the event really shitty.
* SMES do NOT lose charge, do NOT have their settings fucked with, they just BSOD for as long as the event continues, not charging and not providing power

Fixes #11704
:cl:
 * bugfix: The 'temporary powernet disable' random event no longer magically refills all SMES or sets their output to M A X I M U M after power is restored. It will also not restore lighting/environment/equipment in APCs if they were disabled before the event.

### CODE REVIEW

File: code/game/gamemodes/events/power_failure.dm [hunk 1]
```diff
@@ -54,21 +55,19 @@
 		command_alert(/datum/command_alert/power_restored)
 	for(var/obj/machinery/power/apc/C in power_machines)
 		if(C.cell && C.z == map.zMainStation)
-			C.cell.charge = C.cell.maxcharge
+			C.cell.charge = min(C.cell.maxcharge, max(C.old_charge, C.cell.charge))
```
Comment (ComicIronic) [inline code/game/gamemodes/events/power_failure.dm]: `Clamp(C.cell.charge, C.old_charge, C.cell.maxcharge)`

File: code/modules/power/battery.dm [hunk 2]
```diff
@@ -297,23 +298,10 @@ var/global/list/battery_online =	list(
...
-	var/old_online = online
-	var/old_charging = charging
-	var/old_output = output
-
-	online = 0
-	charging = 0
-	output = 0
 	charge = max(0, charge - 1e6/severity)
-
+	stat |= FORCEDISABLE
```
Comment (ComicIronic) [inline code/modules/power/battery.dm]: There's already a flag for being emp'd, which is what should be used here. I recommend FORCEDISABLE is reserved for events to make it clear what each flag is for.

Comment (9600bauds) [reply code/modules/power/battery.dm]: Good one

**Comment by Sprok0** [comment]
I remember cluster said he fixed this, I watched him stream bugfixes that one time.
#12271 / https://github.com/d3athrow/vgstation13/pull/12271/commits/d15e2850b5e6efc347cadd435d051ddeecc336cb
Some sort of list wizardry I don't understand

**Comment by 9600bauds** [comment]
Oh
it was only fixed for the random event and not the adminbus button

I think?

**Comment by 9600bauds** [comment]
Well that won't be needed anymore

**Comment by Probe1** [comment]
>	out with the old in with the new old
feels like most PRs I merge

**Pull request merged**